tests/Dfe.PlanTech.Data.Sql.UnitTests/Entities/EstablishmentEntityTests.cs

```diff
@@ -0,0 +1,225 @@
+        };
+
+        // Act
+        SqlEstablishmentDto dto = entity.AsDto();
+
+        // Assert - properties explicitly set by `AsDto()`
+        Assert.Equal(expectedId, dto.Id);
+        Assert.Equal(expectedRef, dto.EstablishmentRef);
+        Assert.Equal(expectedType, dto.EstablishmentType);
+        Assert.Equal(expectedOrgName, dto.OrgName);
+        Assert.Equal(expectedGroupUid, dto.GroupUid);
+        Assert.Equal(expectedDateCreated, dto.DateCreated);
+        Assert.Equal(expectedDateLastUpdated, dto.DateLastUpdated);
+
+        // Assert - ensure all DTO properties are accounted for
+        DtoPropertyCoverageAssert.AssertAllPropertiesAccountedFor<SqlEstablishmentDto>(
+            new[]
+            {
+                nameof(SqlEstablishmentDto.Id),
+                nameof(SqlEstablishmentDto.EstablishmentRef),
+                nameof(SqlEstablishmentDto.EstablishmentType),
+                nameof(SqlEstablishmentDto.OrgName),
+                nameof(SqlEstablishmentDto.GroupUid),
+                nameof(SqlEstablishmentDto.DateCreated),
+                nameof(SqlEstablishmentDto.DateLastUpdated)
+            }
+        );
+    }
+
+    [Fact]
+    public void EstablishmentEntity_AsDto_WhenDateCreatedNotProvided_DefaultsToUtcNow()
+    {
+        // Arrange
+        var before = DateTime.UtcNow;
+
+        var entity = new EstablishmentEntity
+        {
+            Id = 1,
+            EstablishmentRef = "Arbitrary string - establishment ref",
+            EstablishmentType = "Arbitrary string - establishment type",
+            OrgName = "Arbitrary string - organisation name"
+            // DateCreated is not set explicitly
+        };
+
+        var after = DateTime.UtcNow;
+
+        // Act
+        SqlEstablishmentDto dto = entity.AsDto();
+
+        // Assert
+        Assert.InRange(dto.DateCreated, before, after);
+        Assert.Equal(DateTimeKind.Utc, dto.DateCreated.Kind);
+    }
+
+    // -------------------------
+    // Property behavior tests
+    // -------------------------
+
+    [Theory]
+    [InlineData(EstablishmentEntity.OrgNameMaxLengthInclusive + 1)] // over max by 1 (boundary)
+    public void EstablishmentEntity_OrgName_WhenInputLengthExceedsMax_AdditionalCharactersTrimmed(int inputLength)
+    {
+        // Arrange
+        string original = new string('O', inputLength);
+
+        // Act
+        var entity = new EstablishmentEntity
+        {
+            OrgName = original,
+            EstablishmentRef = new string('R', EstablishmentEntity.EstablishmentRefMaxLengthInclusive - 1),
+            EstablishmentType = new string('T', EstablishmentEntity.EstablishmentTypeMaxLengthInclusive - 1),
+        };
+
+        // Assert
+        Assert.Equal(EstablishmentEntity.OrgNameMaxLengthInclusive, entity.OrgName.Length);
+        Assert.Equal(original.Substring(0, EstablishmentEntity.OrgNameMaxLengthInclusive), entity.OrgName);
+        Assert.NotEqual(original, entity.OrgName);
+    }
```

Comment:
These tests are adapted from the pre-existing `EstablishmentTests.cs`

It would be interesting to understand the history of how/why we _SILENTLY_ trim these values at all (as opposed to a plain / regular max length and usual associated checks).

Reply:
Created #277225 to track this